invenio_circulation/config.py

```diff
@@ -46,6 +46,11 @@
 Items that have attached loans with these circulation statuses are
 not available to be loaned by patrons."""
 
+CIRCULATION_STATES_LOAN_ACTIVE_EXCEPT = ['ITEM_AT_DESK']
```

Comment:
This solution is because you want to fix the problem with checkout a loan with status 'ITEM_AT_DESK' or because you want to introduce exceptional loan checkouts for specific patrons? Also see the comment from @ntarocco about this bug :)

Reply:
The initial problem is: in some cases (here ITEM_AT_DESK), the loan owner can make a checkout of the item. In invenio-circulation it cannot. This PR attempts to resolve this problem by creating 2 things:
1/ a configuration to list all states in which a loan can be checked-out by it's patron
2/ uses the patron as additional piece of information to determine if the item is considered as available

Reply:
Actually, the fact that the patron must be always the same it is already enforced by the `ensure_same_patron` decorator, see [here](https://github.com/inveniosoftware/invenio-circulation/blob/master/invenio_circulation/transitions/base.py#L166): when a loan is created for the first time, it is created with a `patron_pid`. This cannot be changed while performing a transition. This extra check for patron should **not** be needed.

My guess is that it is still enough to remove the `ITEM_AT_DESK` and transit state from the "active loan" notion: I still think that given that the loan is not really started yet (start date), item at desk should not be considered active loan, but probably something else if really needed.

I will be on holidays for the next 10 days, feel free to ping @zzacharo @kprzerwa @topless!